application/src/main/java/org/togetherjava/tjbot/routines/ModAuditLogRoutine.java

```diff
@@ -0,0 +1,371 @@
+                    guild.getIdLong());
+        });
+
+        logger.info(
+                "Finished checking audit logs of all guilds. The next check is scheduled to be executed in {} hours.",
+                CHECK_AUDIT_LOG_EVERY_HOURS);
+    }
+
+    private void handleAuditLogs(@NotNull MessageChannel auditLogChannel,
+            @NotNull PaginationAction<? extends AuditLogEntry, AuditLogPaginationAction> auditLogAction,
+            long guildId) {
+        @SuppressWarnings("squid:S1602")
+        Instant lastAuditLogEntryTimestamp = database.read(context -> {
+            return Optional
+                .ofNullable(context.fetchOne(ModAuditLogGuildProcess.MOD_AUDIT_LOG_GUILD_PROCESS,
+                        ModAuditLogGuildProcess.MOD_AUDIT_LOG_GUILD_PROCESS.GUILD_ID.eq(guildId)))
+                .map(entry -> entry.get(
+                        ModAuditLogGuildProcess.MOD_AUDIT_LOG_GUILD_PROCESS.LAST_PROCESSED_AUDIT_LOG_ENTRY))
+                .orElse(Instant.now());
+        });
+        // NOTE This is a minor race condition. By taking the time before the actual lookup we
+        // ensure that we do not miss anything but instead it is possible to receive an
+        // action twice in such a rare case, which is okay.
+        Instant updatedLogEntryTimestamp = Instant.now();
+
+        // All entries after last lookup, chronologically ascending
+        auditLogAction.stream()
+            .takeWhile(entry -> isSnowflakeAfter(entry, lastAuditLogEntryTimestamp))
+            .sorted(Comparator.comparing(TimeUtil::getTimeCreated))
+            .map(entry -> handleAuditLog(auditLogChannel, entry))
+            .flatMap(Optional::stream)
+            .reduce((firstMessage, secondMessage) -> firstMessage.flatMap(result -> secondMessage))
+            .ifPresent(RestAction::queue);
+
+        database.write(context -> {
+            var entry = context.newRecord(ModAuditLogGuildProcess.MOD_AUDIT_LOG_GUILD_PROCESS);
+            entry.setGuildId(guildId);
+            entry.setLastProcessedAuditLogEntry(updatedLogEntryTimestamp);
+
+            if (entry.update() == 0) {
+                entry.insert();
+            }
+        });
+    }
+
+    private Optional<RestAction<Message>> handleAuditLog(@NotNull MessageChannel auditLogChannel,
+            @NotNull AuditLogEntry entry) {
+        Optional<RestAction<MessageEmbed>> maybeMessage = switch (entry.getType()) {
+            case BAN -> handleBanEntry(entry);
+            case UNBAN -> handleUnbanEntry(entry);
+            case KICK -> handleKickEntry(entry);
+            case MEMBER_ROLE_UPDATE -> handleRoleUpdateEntry(entry);
+            case MESSAGE_DELETE -> handleMessageDeleteEntry(entry);
+            default -> Optional.empty();
+        };
+        return maybeMessage.map(message -> message.flatMap(auditLogChannel::sendMessageEmbeds));
+    }
+
+    @SuppressWarnings("MethodMayBeStatic")
```

Comment:
I'd make the methods static myself, like IntelliJ is saying

Reply:
I think this is a design flaw. As a consequence, all of those methods should be static but the role ones cant be static...
I think those methods are non-static by design but could be made static as of today, due to their current implementation.

I dont have a strong voice here though. I can make it static.

Reply:
If only a portion of the methods can become static, I'm fine with keeping them all instance because of consistency reasoning.

If they all can get static, I'd for sure make them all static.